indico/modules/auth/util.py

```diff
@@ -69,3 +70,12 @@ def url_for_login(next_url=None):
 
 def url_for_logout(next_url=None):
     return url_for('auth.logout', next=next_url)
+
+
+def url_for_register(next_url=None):
+    local_identities_enabled = Config.getInstance().getLocalIdentities()
+    if local_identities_enabled:
+        return url_for('auth.register', next=next_url)
+
+    external_url = Config.getInstance().getExternalRegistrationUrl()
```

Comment:
An IDEs wouldn't really help here; this is how those ugly getters are implemented:

    def __getattr__(self, attr):
        if attr[0:3] == 'get':
            def configFinder(k):
                return self._configVars[k]
            return lambda: configFinder(attr[3:])
        else:
            raise AttributeError